### ISSUE TITLE
[Tablet Orders] Cell selection color in dark mode is not visible 

### ISSUE THREAD

**Issue by iamgabrielma** [woocommerce/woocommerce-ios#11768]
In dark mode the cell selection color is not very perceptible as does not have so much contrast compared with light mode:

<img width=600 src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/71fff83a-5e08-4ace-a5bb-5deb394f789d"/>

**Comment by joshheald** [comment]
This seems like a change. Previously, I saw the selected cells the black colour that you're seeing for every cell, and the unselected cells were the same colour as the cells in the detail view.

**Comment by joshheald** [comment]
I can't repro this, the cells are as I described in my previous comment. LMK if you still can see this in `trunk`....

**Comment by bozidarsevo** [comment]
> I can't repro this

Same for me:


![Image](https://github.com/woocommerce/woocommerce-ios/assets/6242034/dc014a74-387b-4048-996e-79d8454d83cd)

**Comment by iamgabrielma** [comment]
> I saw the selected cells the black colour that you're seeing for every cell, and the unselected cells were the same colour as the cells in the detail view.

Yes, from your comment and checking @bozidarsevo screenshot I see there are other differences as well in the cells: On my device the cells are black, then the selected cell is purple, while in your device these appear as grey, and the selected cell is black.

This however, only seems to happen on my physical device running on 15.3.1, not on the simulator running 16.4, so I thought is possible that comes from the cell configuration when we override `updateConfiguration(using state: UICellConfigurationState)`, since we do an iOS16 check there in order to use the default background configuration or not.

I'm not quite sure why this is working correctly in dark mode on iOS16+ (perhaps cell config magic?), but it should display wrongly (as in my physical device) since [this seems to be a bug](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Orders/Cells/OrderTableViewCell.swift#L111-L114) as the  `.wooCommercePurple(.shade0)` would be applied on light and dark mode to selected cells:

```
        if state.isSelected || state.isHighlighted {
            backgroundConfiguration.backgroundColor = .wooCommercePurple(.shade0)
        }
```
Currently do not have a `SemanticColor` that uses `wooCommercePurple(.shade0)` in light mode and a different one in dark mode. If I change the above to use something like the following, then it seems to work fine (we would need another one for unselected cells I'm guessing):

```
    static var selectableSomething: UIColor {
        return UIColor(light: .wooCommercePurple(.shade0), dark: .black)
    }
```

**Comment by joshheald** [comment]
> > I saw the selected cells the black colour that you're seeing for every cell, and the unselected cells were the same colour as the cells in the detail view.
> 
> Yes, from your comment and checking @bozidarsevo screenshot I see there are other differences as well in the cells: On my device the cells are black, then the selected cell is purple, while in your device these appear as grey, and the selected cell is black.
> 
> This however, only seems to happen on my physical device running on 15.3.1, not on the simulator running 16.4, so I thought is possible that comes from the cell configuration when we override `updateConfiguration(using state: UICellConfigurationState)`, since we do an iOS16 check there in order to use the default background configuration or not.

Yeah, I can repro these issues on my iOS 15 device too.

Since the difference comes on iOS 15, it's probably to do with the version-specific init we use. I took a look at [the docs for UIBackgroundConfiguration](https://developer.apple.com/documentation/uikit/uibackgroundconfiguration#) and noticed that one of the options is `listGroupedCell`. I hadn't given `listPlainCell` much thought, but the order list is actually a grouped table.

Changing to `listGroupedCell` seems to fix it on iOS 15 – see this PR: #11786

> I'm not quite sure why this is working correctly in dark mode on iOS16+ (perhaps cell config magic?), but it should display wrongly (as in my physical device) since [this seems to be a bug](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Orders/Cells/OrderTableViewCell.swift#L111-L114) as the `.wooCommercePurple(.shade0)` would be applied on light and dark mode to selected cells:
> 
> Currently do not have a `SemanticColor` that uses `wooCommercePurple(.shade0)` in light mode and a different one in dark mode.

`woocommercePurple(.shade0)` does this itself:

[`UIColor+SemanticColors`](https://github.com/woocommerce/woocommerce-ios/blob/be6c4314549499c4acc48411bfb86b496ee5202f/WooFoundation/WooFoundation/Colors/UIColor%2BSemanticColors.swift#L500)
```
case .shade0:
            return UIColor(light: .withColorStudio(.wooCommercePurple, shade: .shade0), dark: .withColorStudio(.wooCommercePurple, shade: .shade100))
```
